ui/utils.py

```diff
@@ -210,3 +210,46 @@ def tick(self):
 
 
 Rect = namedtuple('Rect', ['left', 'top', 'right', 'bottom'])
+
+def fit_image_to_screen(image, o):
+    """Fits a given image to fit on any sized screen whilst maintaining the aspect ratio
+
+    Args:
+
+        * ``image``: Specify a PIL image here
+        * ``o``: output device object."""
+
+    image_width, image_height = image.size
+    if o.height > image_height and o.width > image.width: # Checks if the screen dimensions are equal to the image size
+	logger.info("Using the resize script")
+        if o.height/image_height < o.width/image_width: # Checks which side is bigger in proportion to the image size
+            bigger_side = o.height
+            bigger_image_side = image_height
+            smaller_image_side = image_width
+        else:
+            bigger_side = o.width
+            bigger_image_side = image_width
+            smaller_image_side = image_height
+        bigger_side_percent = (bigger_side/float(bigger_image_side))
+        other_size = int((float(smaller_image_side)*float(bigger_side_percent))) # Working out smaller side length
+        image = image.resize((bigger_side,other_size), Image.ANTIALIAS) # Resizes the image to the calculated dimensions to fit the screen and stick to the aspect ratio using a $
+    elif (o.width, o.height) == image.size: # Checks if screen dimensions and exactly the same as image dimensions
+	pass
+    elif (o.width == image_width and o.height > image_height) or (o.height == image_height and o.height > image_height):
```

Comment:
`(o.height == image_height and o.height > image_height)` doesn't make sense and will never be `True`. If this makes something fail, add a testcase that checks for it =)

Reply:
Doh, will double check I have a tests for every scenario

Reply:
And of course fix it